src/sql/src/parsers/copy_parser.rs

```diff
@@ -39,10 +41,74 @@ impl<'a> ParserContext<'a> {
                     actual: self.peek_token_as_string(),
                 })?;
 
-        self.parser
-            .expect_keyword(Keyword::TO)
+        let direction = self
+            .parser
+            .expect_one_of_keywords(&[Keyword::TO, Keyword::FROM])
+            .context(error::SyntaxSnafu { sql: self.sql })?;
+
+        match direction {
+            Keyword::TO => self.parse_copy_table_to(table_name),
+            Keyword::FROM => self.parse_copy_table_from(table_name),
+            _ => unreachable!(),
+        }
+    }
+
+    fn parse_copy_table_from(&mut self, table_name: ObjectName) -> Result<CopyTable> {
+        let uri = self
+            .parser
+            .parse_literal_string()
+            .with_context(|_| error::UnexpectedSnafu {
+                sql: self.sql,
+                expected: "a uri",
+                actual: self.peek_token_as_string(),
+            })?;
+
+        let options = self
+            .parser
+            .parse_options(Keyword::WITH)
             .context(error::SyntaxSnafu { sql: self.sql })?;
 
+        // default format is parquet
+        let mut format = Format::Parquet;
+        let mut pattern = None;
+        for option in options {
+            match option.name.value.to_uppercase().as_str() {
+                "FORMAT" => {
+                    if let Value::SingleQuotedString(fmt_str) = option.value {
+                        format = Format::try_from(fmt_str)?;
+                    }
```

Comment:
Shouldn't we return error when `option.value` isn't a string? Could we also support double-quoted strings?